cmd/kube-apiserver/app/options/completion.go

```diff
@@ -71,6 +72,7 @@ func (opts *ServerRunOptions) Complete() (CompletedOptions, error) {
 	completed.PrimaryServiceClusterIPRange = primaryServiceIPRange
 	completed.SecondaryServiceClusterIPRange = secondaryServiceIPRange
 	completed.APIServerServiceIP = apiServerServiceIP
+	completed.FlagsReader = flagz.NamedFlagSetsReader{FlagSets: opts.Flags()}
```

Comment:
this looks misplaced. Would expect it in pkg/controlplane/apiserver/options/options.go.

Reply:
I want to be able to call ServerRunOptions's Flags() via the `opts.Flags()` call to retrieve the flagsets which I dont think I can call from pkg/controlplane/apiserver/options/options.go ?

Reply:
I see what you mean. if this is something in a lower layer (generic controlplane or even apiserver) I would expect it to be wired by default, i.e. the author consuming generic controlplane or apiserver should be forced to wire it.

Thinking about something like:
```
controlplane, err := opts.Options.Complete(opts.Flags(), []string{"kubernetes.default.svc", "kubernetes.default", "kubernetes"}, []net.IP{apiServerServiceIP})
```
i.e. to pass it down as a mandatory argument of the completion chain.

Reply:
I see. Made it a mandatory argument thats passed down in `Options.Complete()` in pkg/controlplane/apiserver/options/options.go